src/bson/bson.c

```diff
@@ -2728,9 +2728,16 @@ _bson_as_json_visit_regex (const bson_iter_t *iter,
 {
    bson_json_state_t *state = data;
    const char *c;
+   char *escaped_regex;
+
+   escaped_regex = bson_utf8_escape_for_json (v_regex, -1);
```

Comment:
Just to clarify. The escaping done in `bson_utf8_escape_for_json` includes escaping of "optional characters".

The bson corpus tests suite has a test that expects the optional characters *not* to be escaped.

I read this as both libbson and the bson corpus are both correct.

@xdg How do you suggest we handle it? Duplicate the corpus test and modify it?
Add a test case to the corpus, and mark then mutually exclusive?